src/sagemaker_huggingface_inference_toolkit/transformers_utils.py

```diff
@@ -231,15 +231,18 @@ def infer_task_from_model_architecture(model_config_path: str, architecture_inde
         architecture = config.get("architectures", [None])[architecture_index]
 
     task = None
+    if "HF_TASK" in os.environ:
+        return os.environ["HF_TASK"]
+
     for arch_options in ARCHITECTURES_2_TASK:
         if architecture.endswith(arch_options):
             task = ARCHITECTURES_2_TASK[arch_options]
 
     if task is None:
         raise ValueError(
-            f"Task couldn't be inferenced from {architecture}."
-            f"Inference Toolkit can only inference tasks from architectures ending with {list(ARCHITECTURES_2_TASK.keys())}."
-            "Use env `HF_TASK` to define your task."
+            f"Task not supported via {list(SUPPORTED_TASKS.keys())} or"
+            f" {list(ARCHITECTURES_2_TASK.keys())}."
+            "Use inference.py to install unsupported task separately"
```

Comment:
I think all validations should take place in the handler_service https://github.com/aws/sagemaker-huggingface-inference-toolkit/blob/main/src/sagemaker_huggingface_inference_toolkit/handler_service.py#L101. Just update the error messaging here

Reply:
Thanks updated